### ISSUE TITLE
Allow to use SQLite or PostgreSQL

### ISSUE THREAD

**Issue by jeffbicca** [uptrace/uptrace#217]
I know there is [this](https://github.com/uptrace/uptrace/pull/102) that "solves" [this](https://github.com/uptrace/uptrace/issues/90) but I still think that having support for both databases would be important.

Since we are talking about a database that it's use case is for Uptrace's own data and besides that SQLite and PostgreSQL each have it's own set of advantages and disadvantages, in my opinion it would be great have the option depending of the architecture and context that Uptrace is getting deployed.

Maybe let PostgreSQL be the default option and those who need a simpler architecture could change the uptrace.yml config file just like it was in previous versions of Uptrace.

**Comment by rattuscz** [comment]
Definitely would love to have option to still use sqlite, as for our usecase there will be single instance and postgresql only adds additional dependency and complexity for deployment and maintenance.

**Comment by vmihailenco** [comment]
Yes, having SQLite support would be nice, but it is quite some additional work and some features like [faceted navigation](https://bun.uptrace.dev/postgres/faceted-full-text-search-tsvector.html) in [alerts](https://app.uptrace.dev/alerting/1/alerts) just can't be easily ported...

**Comment by jeffbicca** [comment]
> Yes, having SQLite support would be nice, but it is quite some additional work and some features like [faceted navigation](https://bun.uptrace.dev/postgres/faceted-full-text-search-tsvector.html) in [alerts](https://app.uptrace.dev/alerting/1/alerts) just can't be easily ported...

Hi @vmihailenco .Thank you very much for your reply.  So in Uptrace there are already a few features that uses the advantages that PostgreSQL offers. That's fine!

I don't know about the whole Uptrace's code base and features as I am a first time user of it and I still experimenting it but for this specific feature I could see a feature toggle like "if Uptrace is configured to use SQLite, the faceted navigation in alerts would be unavailable". I don't know if there would be any more other cases.

BTW In the systems where I do work in case of having different and lots of alerts that for sure would means trouble for me, because it's a sign that the system is not behaving as it would be. :rofl: 
So the search and navigation in these situations would not be used as the ideal situation is to have just a few dozen of alerts in max.

Still, Uptrace is looking promising in my company use case because from what I've been messing around with other solutions it's the one that requires less of external infrastructure and if it's needed to scale it seems quite easier than in any other software.

Thank you very much for it!
